libdino/src/service/secret_manager.vala

```diff
@@ -29,15 +30,18 @@ public class SecretManager {
         }
     }
 
-    public void set_password(string password) {
+    public async bool set_password(string password) {
         try {
-            Secret.password_store_sync(schema, Secret.COLLECTION_DEFAULT,
+            return yield Secret.password_store(schema, Secret.COLLECTION_DEFAULT,
                                        "XMPP account " + user + "@" + server, password, null,
                                        "user", user,
                                        "server", server,
                                        "protocol", "xmpp");
-        } catch (GLib.Error e) { }
+        } catch (GLib.Error e) {
```

Comment:
Would it make sense to propagate this error to the caller, so they could report the full error to the user?

Reply:
I have no idea. I don't even know under which conditions an error is raised here; the libsecret documentation does not mention this at all.